apis/python/src/tiledbsoma/util_tiledb.py

```diff
@@ -96,3 +100,114 @@ def is_duplicate_group_key_error(e: tiledb.TileDBError) -> bool:
         return True
 
     return False
+
+
+_TDBO = TypeVar("_TDBO", bound=tiledb.Object)
+_Self = TypeVar("_Self")
+
+
+@attrs.define()
+class ReadWriteHandle(Generic[_TDBO]):
+    """Paired read/write handles to a TileDB object.
+
+    You can (and should) access the members, but do not write to them.
+    """
+
+    uri: str
+    context: SOMATileDBContext
+    read: _TDBO
```

Comment:
Very awkwardly named. `read` is a verb and a noun and an adjective; and a Python builtin.

Please call this something (anything) else: maybe `object_for_read`, or `read_object`, ...

Reply:
These are now `reader` and `writer`.